feat(promtail): add more complex options of dropping logs at the drop stage (#6371)
grafana/loki: docs/sources/upgrading/_index.md

This value now defaults to `loki`, it was previously set to `cortex`. If you are relying on this container name for your chunks or ruler storage, you will have to manually specify `-azure.container-name=cortex` or `-ruler.storage.azure.container-name=cortex` respectively.

### Promtail

#### Drop stages config changed

> Config option `source` was renamed to `sources` and now acts as a list of strings. Option `value` was removed and use 

Reviewer: ```suggestion
The configuration option `source` was renamed to `sources` and now accepts a list of strings. 
```

> option `expression` as regexp instead of. 

Reviewer: ```suggestion
Also, the configuration option `value` was removed in favor of the option `expression`, which takes a regular expression. 
```
